caffe2/core/operator_c10wrapper.h

```diff
@@ -25,104 +16,100 @@ using extract_type_t = typename ParameterDef::type;
  *
  */
 
-template <
-    const c10::OperatorHandle& (*OperatorHandle)(),
-    class Context,
-    bool use_array_input,
-    size_t num_output_parameters,
-    class ParameterDefTuple>
+namespace detail {
+template <class Context>
 class C10OperatorWrapper final : public Operator<Context> {
  public:
-  static_assert(
-      c10::guts::is_instantiation_of<std::tuple, ParameterDefTuple>::value,
-      "");
-  using ParameterTuple =
-      c10::guts::typelist::to_tuple_t<c10::guts::typelist::map_t<
-          details::extract_type_t,
-          c10::guts::typelist::from_tuple_t<ParameterDefTuple>>>;
-
   USE_OPERATOR_CONTEXT_FUNCTIONS;
 
-  C10OperatorWrapper(const OperatorDef& operator_def, Workspace* ws)
+  C10OperatorWrapper(
+      const c10::OperatorHandle& op,
+      const OperatorDef& operator_def,
+      Workspace* ws)
       : Operator<Context>(operator_def, ws),
-        op_(OperatorHandle()),
+        op_(op),
         kernel_(at::nullopt),
-        parameters_(parse_parameters_(
-            operator_def,
-            c10::guts::make_index_sequence<num_parameters()>())) {
+        has_preallocated_outputs_(
+            op_.schema().arguments().back().name() ==
```

Comment:
So... invariant, you cannot have a C10OperatorWrapper for a schema with no arguments? (Aren't there some Caffe2 operators which don't have arguments?)

Reply:
good catch